src/core/utils/number/numberUtils.ts

```diff
@@ -0,0 +1,99 @@
+import {
+  IS_LAST_CHARACTER_DECIMAL_POINT_REGEX,
+  NOT_NUMBER_OR_DECIMAL_POINT_REGEX
+} from "./numberConstants";
+import {FormatNumberOptions, ParseNumberOptions} from "./numberTypes";
+
+function formatNumber({providedOptions}: FormatNumberOptions) {
+  const {locale, ...otherOptions} = providedOptions;
+  const options = {
+    style: "decimal",
+    ...otherOptions
+  };
+
+  let numberFormatter: {
+    format: (x: number | bigint) => string;
+  };
+
+  try {
+    numberFormatter = new Intl.NumberFormat(
+      locale || [navigator.language, "en-GB"],
+      options
+    );
+  } catch (error) {
+    numberFormatter = {
+      format(x: number | bigint) {
+        return x.toLocaleString(locale);
+      }
+    };
+  }
+
+  return (value: number) => {
+    let formattedValue = "";
+
+    if (!Object.is(value, NaN)) {
+      formattedValue = numberFormatter.format(value);
+    }
+
+    return formattedValue;
+  };
+}
+
+/**
+ * Coerces a number scientific notation. {@link https://observablehq.com/@mbostock/localized-number-parsing | Reference}
+ * @param {object} options - An object includes parse number options
+ * @param {string} options.locale - Default locale used is browser locale
+ * @param {number} options.maximumFractionDigits - The maximum digit a decimal can have
+ * @param {number} value - A number to convert to string
+ * @returns {string} The value after coercing the given value to a scientific notation.
+ */
+function parseNumber(
+  options: ParseNumberOptions = {locale: navigator.language, maximumFractionDigits: 0},
+  value: string
+) {
+  const {THOUSANDTHS_SEPARATOR, DECIMAL_NUMBER_SEPARATOR} = getNumberSeparators(
+    options.locale
+  );
+  const numerals = [
+    // eslint-disable-next-line no-magic-numbers
+    ...new Intl.NumberFormat(options.locale, {useGrouping: false}).format(9876543210)
+  ].reverse();
+  const numeral = new RegExp(`[${numerals.join("")}]`, "g");
+  const digitMapper = getDigit(new Map(numerals.map((d, i) => [d, i])));
+  let parsedNumber = value
+    .replace(new RegExp(NOT_NUMBER_OR_DECIMAL_POINT_REGEX), "")
+    .replace(new RegExp(`[${THOUSANDTHS_SEPARATOR}]`, "g"), "")
+    .replace(new RegExp(`[${DECIMAL_NUMBER_SEPARATOR}]`), ".")
+    .replace(numeral, digitMapper);
+
+  if (options.maximumFractionDigits > 0) {
+    const decimalPart = parsedNumber.split(DECIMAL_NUMBER_SEPARATOR)[1];
+
+    if (decimalPart && decimalPart.length === options.maximumFractionDigits + 1) {
+      return parsedNumber.slice(0, parsedNumber.length - 1);
+    }
+  } else {
+    parsedNumber = parsedNumber.replace(IS_LAST_CHARACTER_DECIMAL_POINT_REGEX, "");
```

Comment:
I think here we want to make sure the parsed number is an integer so I wonder if we can do this instead:
```suggestion
    parsedNumber = String(parseInt(parsedNumber));
```